pkgs/top-level/all-packages.nix

```diff
@@ -8694,11 +8694,12 @@ in
   gerbil-support = callPackage ../development/compilers/gerbil/gerbil-support.nix { };
   gerbilPackages-unstable = gerbil-support.gerbilPackages-unstable; # NB: don't recurseIntoAttrs for (unstable!) libraries
 
-  gccFun = callPackage (if (with stdenv.targetPlatform; isVc4 || libc == "relibc")
-    then ../development/compilers/gcc/6
-    else ../development/compilers/gcc/9);
-  gcc = if (with stdenv.targetPlatform; isVc4 || libc == "relibc")
-    then gcc6 else gcc9;
+  gccFun = callPackage (if stdenv.targetPlatform.isZ80
+                        then ../development/compilers/z88dk
+                        else (if (with stdenv.targetPlatform; isVc4 || libc == "relibc")
+                              then ../development/compilers/gcc/6
+                              else ../development/compilers/gcc/9));
+  gcc = if stdenv.targetPlatform.isZ80 then z88dk else if (with stdenv.targetPlatform; isVc4 || libc == "relibc") then gcc6 else gcc9;
```

Comment:
Since `z88dk` is not a GCC fork, instead of changing this stuff can you change the compiler selected in the stdenv? See https://github.com/NixOS/nixpkgs/blob/412c9c883bbba8553f6c5e33cd8e7a985d05bba3/pkgs/stdenv/cross/default.nix#L56-L70